dali/operators/reader/loader/fits_loader.h

```diff
@@ -58,6 +58,11 @@ class FitsLoader : public FileLoader<Backend, Target> {
 
     DALI_ENFORCE(hdu_indices_.size() == dtypes_.size(),
                  "Number of extensions does not match the number of provided types");
+
+    DALI_ENFORCE(fits_is_reentrant(),
+                 "Loaded instance of cfitsio library does not support multithreading. "
+                 "Please recompile cfitsio in reentrant mode (--enable-reentrant) "
+                 "or use cfitsio delivered in DALI_deps");
```

Comment:
Couldn't we just warn the user and guard the code with a mutex, like this?
```C++
std::unique_lock lock(mtx, std::defer_lock);
if (!fits_is_reentrant)
    lock.lock();
```

Reply:
Good idea! Thank you!

Reply:
I added locking to `FITS_CALL`